File: 01/007/main.ts
function isPrime(number: number) {
  let limit = number;
  for (let i = 2; i < limit; i++) {
    if (number % i === 0) {
      return false;
    } else {
      limit = Math.floor(number / i);
    }
  }
  return true;
}

function primeAt(at: number) {
  let loopen = 0,
    number = 1;
  while (loopen < at) {
    number++;
    if (isPrime(number)) {
      loopen++;
    }
  }
  return number;
}

console.log(primeAt(10001));

File: 01/007/main.rb
def isPrime number
    limit = number
    i = 2
    while i < limit do
        if number % i == 0 then
            return false
        else
            limit = number / i.to_i
        end
        i = i + 1
    end
    true
end

def primeAt at
    loopen, number = 0, 1
    while loopen < at do
        number = number + 1
        if isPrime(number) then
            loopen = loopen + 1
        end
    end
    number
end

puts primeAt(10001)
